std/stdio.d

```diff
@@ -2808,6 +2808,90 @@ is empty, throws an `Exception`. In case of an I/O error throws
         assert(i == witness.length);
     }
 
+/**
+Like `asCharsRange` but returns a range o `ubyte`s, not `char`s.
+*/
+    auto asUbytesRange(size_t bufferSize = 4096)
+    {
+        assert(bufferSize > 0);
+
+        import std.algorithm.iteration;
+        return byChunk(bufferSize).joiner;
+    }
+
+/**
+Returns an $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
+set up to read from the file handle a character at a time.
+
+The element type for the range will be `char`. Range primitives
+may throw `StdioException` on I/O error.
+
+Example:
+---------
+void main()
+{
+    // Read standard input
+    foreach (char buffer; stdin.asCharsRange)
+    {
+        ... use buffer ...
+    }
+}
+---------
+
+The parameter may be a number (as shown in the example above) recommending the
```

Comment:
"recommending" seems to be misleading as it implies that the value may not be followed, but I don't think there are any circumstances where the "recommendation" is ignored (other than there being less that the indicated number of bytes available for reading at that time).

Reply:
For forward compatibility, we need to mark it as recommended, because a future implementation may for example reduce the buffer size if the available memory is low.

Reply:
Sorry, that makes no sense to me. You are overdesigning.